src/Summoner/Config.hs

```
@@ -0,0 +1,170 @@
+    <*> decision       "exe"         .= cExe
+    <*> decision       "test"        .= cTest
+    <*> decision       "bench"       .= cBench
+    <*> lastP (Toml.arrayOf ghcVerV) "ghcVersions" .= cGhcVer
+  where
+    lastP :: (Key -> BiToml a) -> Key -> BiToml (Last a)
+    lastP f = dimapBijection getLast Last . Toml.maybeP f
+
+    ghcVerV :: Valuer 'TString GhcVer
+    ghcVerV = Valuer (matchText >=> parseGhcVer) (Toml.String . showGhcVer)
+
+    license :: Key -> BiToml License
+    license =  bijectionMaker "License" (matchText >=> Just . License) (Toml.String . unLicense)
+
+    decision :: Key -> BiToml Decision
+    decision = dimapBijection fromDecision toDecision . Toml.maybeP Toml.bool
+
+    decisionMaybe :: [(Decision, Maybe Bool)]
+    decisionMaybe = [ (Idk, Nothing)
+                    , (Yes, Just True)
+                    , (Nop, Just False)
+                    ]
+
+    fromDecision :: Decision -> Maybe Bool
+    fromDecision d = snd $ head $ filter ((== d) . fst) decisionMaybe
+
+    toDecision :: Maybe Bool -> Decision
+    toDecision m = fst $ head $ filter ((== m) . snd) decisionMaybe
+
+
+-- | Make sure that all the required configurations options were specified.
+finalise :: PartialConfig -> Either Text Config
+finalise Config{..} = Config
+    <$> fin  "owner"    cOwner
+    <*> fin  "fullName" cFullName
+    <*> fin  "email"    cEmail
+    <*> fin  "license"  cLicense
+    <*> pure cGitHub
+    <*> pure cTravis
+    <*> pure cAppVey
+    <*> pure cPrivate
+    <*> pure cScript
+    <*> pure cLib
+    <*> pure cExe
+    <*> pure cTest
+    <*> pure cBench
+    <*> fin  "ghcVersions" cGhcVer
+  where
+    fin name = maybe (Left $ "Missing field: " <> name) Right . getLast
+
+-- | Read configuration from the given file and return it in data type.
+loadFileConfig :: MonadIO m => FilePath -> m (Either Toml.EncodeException PartialConfig)
+loadFileConfig filePath = liftIO $ TIO.readFile filePath >>= pure . Toml.encode configT
```

I think you can just use `throwIO` here instead of returning `Either`. If there's error in configuration we don't want to skip this file, we want to throw exception. This also will simplify code a bit in `main`.